### PR TITLE
Make SDF filter configurable during session creation

### PR DESCRIPTION
**Pull request by EmanueleGallone**
* Change `nb-addr` flag in `gnb-addr`
* Make SDF Filter empty by default

### CODE REVIEW

File: cmd/pfcpctl/main.go [hunk 1]
```diff
@@ -47,6 +47,7 @@ func main() {
 
 	bufferFlag := getopt.BoolLong("buffer", 'b', "If set, downlink FARs will have the buffer flag set to true")
 	notifyCPFlag := getopt.BoolLong("notifycp", 'm', "If set, downlink FARs will have the notify CP flag set to true")
+	setAppFilterSDF := getopt.BoolLong("no-wildcard", 'w', "If set, downlink FARs will have a SDF filter with app filtering ports")
```
Comment (EmanueleGallone) [inline cmd/pfcpctl/main.go:50]: Ideas on how to name this flag? `app-filter-SDF` ?

File: cmd/pfcpctl/main.go [hunk 2]
```diff
@@ -47,6 +47,7 @@ func main() {
 
 	bufferFlag := getopt.BoolLong("buffer", 'b', "If set, downlink FARs will have the buffer flag set to true")
 	notifyCPFlag := getopt.BoolLong("notifycp", 'm', "If set, downlink FARs will have the notify CP flag set to true")
+	setAppFilterSDF := getopt.StringLong("filter-SDF", 'f', "Allows to set a custom SDF filter to override wildcard SDF filter, on session creation")
```
Comment (daniele-moro) [inline cmd/pfcpctl/main.go:50]:
```suggestion
	setAppFilterSDF := getopt.StringLong("filter-SDF", 'f', "Allows to set a custom SDF filter to override wildcard SDF filter, on session creation, default is wildcard")
```

File: cmd/pfcpctl/main.go [hunk 3]
```diff
@@ -96,6 +97,7 @@ func main() {
 			BaseID:        int32(*baseId),
 			NodeBAddress:  *nodeBAddr,
 			UeAddressPool: *ueAddrPool,
+			FilterSDF: *setAppFilterSDF,
```
Comment (daniele-moro) [inline cmd/pfcpctl/main.go:100]:
```suggestion
			FilterSDF:     *setAppFilterSDF,
```

**Review by daniele-moro — APPROVED** [review]

**Review by osinstom — COMMENTED** [review]
The current proposal assumes "wildcard" as default filter. However, we should be able to send PDRs without any SDF filter, since it's optional (AFAIK). 

Also, I suggest using `--sdf-filter`.

**Comment by EmanueleGallone** [comment]
> * can the SDF filter be modified in the PFCP Session Modification? If not, I propose to somehow disable that.

SDF filter is part of PDR.
Right now, in pfcpsim we don't send an Update PDR IE (The only place where we can actually change the SDF filter).

File: cmd/pfcpctl/main.go [hunk 4]
```diff
@@ -42,11 +42,12 @@ func main() {
 	baseId := getopt.IntLong("baseID", 'i', 1, "First ID used to generate all other ID fields.")
 	n3Addr := getopt.StringLong("n3-addr", 'a', "", "The IPv4 address of the UPF's N3 interface")
 	ueAddrPool := getopt.StringLong("ue-pool", 'u', "", "The IPv4 prefix from which UE addresses will be drawn.")
-	nodeBAddr := getopt.StringLong("nb-addr", 'g', "", "The IPv4 address of the NodeB")
+	nodeBAddr := getopt.StringLong("gnb-addr", 'g', "", "The IPv4 address of the NodeB")
```
Comment (EmanueleGallone) [inline cmd/pfcpctl/main.go:45]: https://github.com/omec-project/up4/pull/237#discussion_r810958680

**Review by osinstom — APPROVED** [review]
LGTM

**Pull request merged**